src/DateInput/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useState } from 'react'
+import Input from '../Input'
+import Select from '../SelectBox'
+import styled from 'styled-components'
+import { isValidDate } from './helpers/validate'
+import { GridInput, Wrapper, Label, Error } from './DateInputStyle'
+
+interface Month {
+  value: number
+  label: string
+}
+
+interface InputLabels {
+  day: string
+  month: string
+  year: string
+}
+
+interface DateInputProps {
+  onChange?: any
+  error?: string | boolean
+  months?: Array<Month>
+  inputLabels?: InputLabels
+  label?: any
+  dateFormatError?: string
+}
+
+const DateInput = ({
+  onChange,
+  error,
+  months,
+  inputLabels,
+  label,
+  dateFormatError
+}: DateInputProps) => {
+  const [value, setValue] = useState({ day: null, month: null, year: null })
+  const [internalError, setInternalError] = useState(null)
+
+  useEffect(() => {
+    const result = {
+      day: value.day ? parseInt(value.day) : null,
+      month: value.month ? value.month.value : null,
+      year: value.year ? parseInt(value.year) : null
+    }
+
+    const allFilled = result.day && result.month && result.year
+
+    if (!isValidDate(result.day, result.month, result.year) && allFilled) {
+      onChange(null)
+      return setInternalError(dateFormatError || 'Date is wrong. Please fix it')
+    }
+
+    if (isValidDate(result.day, result.month, result.year) && allFilled) {
+      setInternalError(null)
+      return onChange(result)
+    } else {
+      return onChange(null)
+    }
+  }, [value])
+
+  const labels = inputLabels || {
+    day: 'Day',
+    month: 'Month',
+    year: 'Year'
+  }
+
+  const DEFAULT_MONTHS = [
+    { value: 1, label: 'January' },
```

Comment:
This does not add up with date validation I mentioned above. Please double check it.